### PR TITLE
added isterable-to-iscollection to registry

### PR DESCRIPTION
**Pull request by mohab-sameh**

### CODE REVIEW

**Comment by mohab-sameh** [comment]
@gregintuitaio should we update the license for this one?

**Comment by grzpab** [comment]
@mohab-sameh can you add some test cases?

File: codemods/jscodeshift/immutable/4/isterable-to-iscollection/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	const root = j(file.source);
+
+	root.find(j.Identifier)
+    .filter(path => path.node.name === 'isIterable')
+      .replaceWith(j.identifier('isCollection'));
+
+	return root.toSource(options);
+};
+
+module.exports.type = 'js';
```
Comment (grzpab) [inline codemods/jscodeshift/immutable/4/isterable-to-iscollection/index.js:42]: please remove this, we use `tsx` everywhere for JSCodeshift codemods

File: codemods/jscodeshift/immutable/4/isterable-to-iscollection/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+module.exports = function transformer(file, api, options) {
+	const j = api.jscodeshift;
+
+	const root = j(file.source);
+
+	root.find(j.Identifier)
+    .filter(path => path.node.name === 'isIterable')
+      .replaceWith(j.identifier('isCollection'));
+
+	return root.toSource(options);
```
Comment (grzpab) [inline codemods/jscodeshift/immutable/4/isterable-to-iscollection/index.js:39]: check the code in this repo for `dirtyFlag`, you can check if the replacing was actually done and only then call `root.toSource()`

File: codemods/jscodeshift/immutable/4/isterable-to-iscollection/index.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/*
+The MIT License (MIT)
+
+Copyright (c) 2023 Rajasegar Chandran
```
Comment (grzpab) [inline codemods/jscodeshift/immutable/4/isterable-to-iscollection/index.js:4]: I think you are the creator of this codemod right?

File: codemods/jscodeshift/immutable/4/isterable-to-iscollection/config.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+	"schemaVersion": "1.0.0",
+	"name": "immutable/4/isterable-to-iscollection",
```
Comment (grzpab) [inline codemods/jscodeshift/immutable/4/isterable-to-iscollection/config.json:3]: you need to add this codemod to the codemod set in the `/sets/` directory.

Comment (mohab-sameh) [reply codemods/jscodeshift/immutable/4/isterable-to-iscollection/config.json:3]: resolved in #25

File: codemods/jscodeshift/immutable/4/isterable-to-iscollection/index.js (hunk 3 above)
Comment (mohab-sameh) [reply codemods/jscodeshift/immutable/4/isterable-to-iscollection/index.js]: resolved in #25

File: codemods/jscodeshift/immutable/4/isterable-to-iscollection/index.js (hunk 2 above)
Comment (mohab-sameh) [reply codemods/jscodeshift/immutable/4/isterable-to-iscollection/index.js]: resolved in #25 in [changed to ts, fixed comments](https://github.com/intuita-inc/codemod-registry/pull/25/commits/5e1f4bac2967053c351570cb8ff711c367c49161)

File: codemods/jscodeshift/immutable/4/isterable-to-iscollection/index.js (hunk 3 above)
Comment (mohab-sameh) [reply codemods/jscodeshift/immutable/4/isterable-to-iscollection/index.js]: resolved in [removed redundant license](https://github.com/intuita-inc/codemod-registry/pull/25/commits/8bbe0b442a9f5ac1521c698a9287e2f7116c8223)

**Review by grzpab — APPROVED** [review]

**Pull request merged**